### ISSUE TITLE
bug: merge into can't found scalar value column in source query

### ISSUE THREAD

**Issue by b41sh** [datafuselabs/databend#13291]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v1.2.161-nightly

### What's Wrong?

merge into can't found scalar value column in source query

### How to Reproduce?

```sql
mysql> create table t11(a int,b string, c string);
Query OK, 0 rows affected (0.37 sec)

mysql> create table t12(a int,b string, c string);
Query OK, 0 rows affected (0.20 sec)

mysql> insert into t11 values(1,'b1','c1'),(2,'b2','c2');
Query OK, 2 rows affected (0.13 sec)

mysql> insert into t12 values(1,'b_5','c_5'),(3,'b_6','c_6');
Query OK, 2 rows affected (0.08 sec)

mysql> merge into t11 using (select a, c, 100 as d from t12) as t112 on t11.a = t112.d  when matched then update set t11.c = t112.c;
ERROR 1105 (HY000): SemanticError. Code: 1065, Text = error:
  --> SQL:1:79
  |
1 | merge into t11 using (select a, c, 100 as d from t12 as t112) on t11.a = t112.d  when matched then update set t11.c = t112.c
  |                                                                               ^ column d doesn't exist
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by b41sh** [comment]
has fixed after #13293 merged
